### PR TITLE
build: Move CI options to CMake presets

### PR DESCRIPTION
**Pull request by nbolton**
Fixes: #7726

This also updates docs, inline with hacking guide, but I have added a TODO to #7673 to reflect any relevant changes.

### CODE REVIEW

**Comment by sithlord48** [comment]
I honestly rather set the cmake options in the clear. The more you hide things the harder it is to follow..

To build we need onlt to run cmake -S. -Bbuild 
everything else set in the preset is constricting ppl to those options. wanna build with ninja on linux TOO BAD the preset sets says you can not.. these over bearing presets are as bad as having config and build scripts. your imposing your environment on others

File: res/dist/arch/PKGBUILD.in [hunk 1]
```diff
@@ -32,7 +32,7 @@ options=('!debug')
 package() {
   # By default, `makepkg` will run from the `src` directory, which would
   # only install the binaries, and not the .desktop file, etc. To install
-  # everything, we need to run `make install` with the root Makefile.
+  # everything, we need to set `--prefix`.
```
Comment (sithlord48) [inline res/dist/arch/PKGBUILD.in:35]: This change is ok

File: CMakePresets.json [hunk 2]
```diff
@@ -36,97 +30,78 @@
...
-      }
-    },
-    {
-      "name": "linux-debug",
-      "inherits": "linux",
+      "inherits": "dev",
+      "generator": "Ninja",
       "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "Debug"
+        "CMAKE_INSTALL_PREFIX": "/usr"
```
Comment (sithlord48) [inline CMakePresets.json:44]: great example something you should not be setting for the same reason we dont set it in the cmake file itself..

**Review by sithlord48 — CHANGES_REQUESTED** [review]
somethings dont belong here (doc changes)

Id rather remove the presets then try to keep to maintain them even more so since our project is so simple to config we really dont' need this at all .

File: CMakePresets.json (hunk 2 above)
Comment (nbolton) [reply CMakePresets.json:44]: I don't think you can compare the cmake config against cmake presets. That's exactly why cmake presets exists; to standardize dev env across the team and across environments (local and CI). 

Presets are optional, cmake config is not. You don't have to use it if you don't want to.

File: README.md [hunk 3]
```diff
@@ -61,55 +61,17 @@ Here are a few ways to join in with the project and get involved:
...
-```
-
-**5. Run**
-```
-./build/bin/deskflow
-```
+## Installing
 
-## How to install (packages)
+Community packages coming soon, for now please use: [Continuous Build](https://github.com/deskflow/deskflow/releases)
```
[the remark this answers is not in the document]
Comment (nbolton) [reply README.md:74]: Ok, will move.

**Comment by nbolton** [comment]
> I honestly rather set the cmake options in the clear. The more you hide things the harder it is to follow..

Using presets isn't about hiding things—it's about standardizing for consistency, especially in environments like CI.

- **CMake config** should have sensible defaults for everyone.
- **CMake presets** should help standardize the dev team's environment.

I can't stress enough how crucial it is to avoid drift between CI and dev environments. Without this standardization, we're seeing the environments diverge, which is becoming a serious headache.

> To build we need onlt to run cmake -S. -Bbuild everything else set in the preset is constricting ppl to those options. 

That’s fine for non-devs, packagers, etc. But as developers, we often need a different setup. I’m not suggesting a one-size-fits-all approach—presets are optional and intended to simplify things if you want to align with CI.

> wanna build with ninja on linux TOO BAD the preset sets says you can not.. these over bearing presets are as bad as having config and build scripts. your imposing your environment on others

Using presets is a choice. I use them because I want to replicate the CI environment. If you don’t care about that, you don’t have to use them—they’re entirely optional.

> Id rather remove the presets then try to keep to maintain them even more so since our project is so simple to config we really dont' need this at all .

I’m definitely against removing them. They save me a lot of time and make me more productive. That said, they are optional, and to show that, I’ve removed them from the documentation and added a `dev-` prefix to each preset name to make their purpose clearer.

**Comment by nbolton** [comment]
While there's some good stuff in here (e.g. using cmake for Arch package build), on the whole the idea no longer fits with our philosophy: [Project Config](https://github.com/deskflow/deskflow/wiki/Project-Config)

I'll pick out the good stuff and put it in a new PR.

For context, here's where I made a 360 on my position about presets: https://github.com/deskflow/deskflow/issues/7726

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -364,12 +363,14 @@ jobs:
...
-        export QT_QPA_PLATFORM=offscreen;
+        ./scripts/install_deps.sh
+        cmake -B build --preset=dev-unix-release
+        cmake --build build -j16
+
+        # Integration tests are flakey by nature, make them optional.
+        export QT_QPA_PLATFORM=offscreen
         ./build/bin/unittests
-        ./build/bin/integtests
+        ./build/bin/integtests || true
```
Comment (nbolton) [inline .github/workflows/ci.yml:373]: Isolated to: #7738

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nbolton) [reply README.md:74]: Moved to #7673

File: BUILD.md [hunk 5]
```diff
@@ -11,6 +11,8 @@ wiki page if you have problems.
 
 **1. Dependencies:**
 
+You can either copy/paste the commands for your OS from [`config.yaml`](config.yaml) or use the deps script.
```
Comment (nbolton) [inline BUILD.md:14]: Moved to #7673

File: res/dist/arch/PKGBUILD.in (hunk 1 above)
Comment (nbolton) [reply res/dist/arch/PKGBUILD.in:35]: Moved to: #7740

**Pull request closed without merging**